university-agent/src/main/java/nl/quintor/studybits/service/AgentService.java

```diff
@@ -61,40 +70,53 @@ else if (messageTypeURN.equals(PROOF.getURN())) {
         throw new NotImplementedException("Processing of message type not supported: " + messageEnvelope.getMessageType());
     }
 
-    public MessageEnvelope<ConnectionResponse> login(String studentId, MessageEnvelope<ConnectionRequest> messageEnvelope) throws IndyException, ExecutionException, InterruptedException, JsonProcessingException {
+    // Student sets up a connection with university agent
+    public MessageEnvelope<ConnectionResponse> login(String studentId, String password, MessageEnvelope<ConnectionRequest> messageEnvelope) throws IndyException, ExecutionException, InterruptedException, JsonProcessingException, AccessDeniedException {
         Student student = studentService.getStudentByStudentId(studentId);
 
+        log.debug("StudentID: "+ studentId);
+        log.debug("STUDENT: " + student);
+        ConnectionRequest connectionRequest = messageEnvelopeCodec.decryptMessage(messageEnvelope).get();
+
+        // If there is no student in the login message, create one for future purposes.
         if (student == null) {
-            student = studentService.createStudent();
+            student = studentService.createStudent(UUID.randomUUID().toString(), UUID.randomUUID().toString(), connectionRequest.getDid());
```

Comment:
It could be null, "". The password can't be null and should at least be an empty string since it will break the passwordEncoder. I made it random to 'really' prove that the authentication fully works without knowing the studentID and Password. It should be never used once there is a connection between agent and student. Even though random is not really necessary to prove this, i thought it was better, i could change this.